scanpy/experimental/pp/_normalization.py

```diff
@@ -54,21 +62,34 @@ def _pearson_residuals(X, theta, clip, check_values, copy: bool = False):
             UserWarning,
         )
 
-    if issparse(X):
-        sums_genes = np.sum(X, axis=0)
-        sums_cells = np.sum(X, axis=1)
-        sum_total = np.sum(sums_genes).squeeze()
+    if not isinstance(X, DaskArray):
+        if issparse(X):
+            sums_genes = np.sum(X, axis=0)
+            sums_cells = np.sum(X, axis=1)
+            sum_total = np.sum(sums_genes).squeeze()
+        else:
+            sums_genes = np.sum(X, axis=0, keepdims=True)
+            sums_cells = np.sum(X, axis=1, keepdims=True)
+            sum_total = np.sum(sums_genes).squeeze()
     else:
-        sums_genes = np.sum(X, axis=0, keepdims=True)
-        sums_cells = np.sum(X, axis=1, keepdims=True)
-        sum_total = np.sum(sums_genes)
-
-    mu = np.array(sums_cells @ sums_genes / sum_total)
-    diff = np.array(X - mu)
+        sums_genes = axis_sum(X, axis=0, dtype=np.float64).reshape(1, -1)
+        sums_cells = axis_sum(X, axis=1, dtype=np.float64).reshape(-1, 1)
+        sum_total = sums_genes.sum()
```

Comment:
These functions should work across the types, no?  They rely on single dispatch of the first argument and just call the other imeplementations